### ISSUE TITLE
Polymorphic type mapping failing to include `stringJsonAdapter`

### ISSUE THREAD

**Issue by pwinckles** [avaje/avaje-jsonb#128]
Hi, I just started trying the library, so I apologize if I missed a step somewhere. The following code is producing an adapter that does not compile:

```java
@Json
@Json.SubType(type = MoveAccuracy.Unavoidable.class, name="UNAVOIDABLE")
@Json.SubType(type = MoveAccuracy.Avoidable.class, name="AVOIDABLE")
public sealed interface MoveAccuracy {
    record Unavoidable() implements MoveAccuracy {}
    record Avoidable(int percentToHit) implements MoveAccuracy {}
}
```

The error is `symbol:   variable stringJsonAdapter` in `MoveAccuracyJsonAdapter`. Here's the generated class:

```java
@Generated
public final class MoveAccuracyJsonAdapter implements JsonAdapter<MoveAccuracy> {

  // naming convention Match
  // percentToHit [int] name:percentToHit ERROR?? no constructor, setter and not a public field? subTypes [com.pwinckles.cassette.common.model.MoveAccuracy.Avoidable]

  private final JsonAdapter<Integer> pintJsonAdapter;
  private final PropertyNames names;

  public MoveAccuracyJsonAdapter(Jsonb jsonb) {
    this.pintJsonAdapter = jsonb.adapter(Integer.TYPE);
    this.names = jsonb.properties("@type", "percentToHit");
  }

  @Override
  public void toJson(JsonWriter writer, MoveAccuracy moveAccuracy) {
    writer.beginObject(names);
    if (moveAccuracy instanceof final MoveAccuracy.Unavoidable sub) {
      writer.name(0);
      stringJsonAdapter.toJson(writer, "UNAVOIDABLE");
    }
    else if (moveAccuracy instanceof final MoveAccuracy.Avoidable sub) {
      writer.name(0);
      stringJsonAdapter.toJson(writer, "AVOIDABLE");
      writer.name(1);
      pintJsonAdapter.toJson(writer, sub.percentToHit());
    }
    writer.endObject();
  }

  @Override
  public MoveAccuracy fromJson(JsonReader reader) {
    // variables to read json values into, constructor params don't need _set$ flags
    int        _val$percentToHit = 0; boolean _set$percentToHit = false;

    String type = null;

    // read json
    reader.beginObject(names);
    while (reader.hasNextField()) {
      final String fieldName = reader.nextField();
      switch (fieldName) {
        case "@type":
          type = stringJsonAdapter.fromJson(reader);
          break;
        case "percentToHit": 
          _val$percentToHit = pintJsonAdapter.fromJson(reader);
          _set$percentToHit = true;
          break;

        default:
          reader.unmappedField(fieldName);
          reader.skipValue();
      }
    }
    reader.endObject();

    if (type == null) {
      throw new IllegalStateException("Missing Required @type property that determines deserialization type");
    }
    if ("UNAVOIDABLE".equals(type)) {
      MoveAccuracy.Unavoidable _$unavoidable = new MoveAccuracy.Unavoidable();
      return _$unavoidable;
    }
    if ("AVOIDABLE".equals(type)) {
      MoveAccuracy.Avoidable _$avoidable = new MoveAccuracy.Avoidable(_val$percentToHit);
      return _$avoidable;
    }
    throw new IllegalStateException("Unknown value for @type property " + type);
  }
}
```

To my eye, this code looks correct, except that it's missing a `private final JsonAdapter<String> stringJsonAdapter` field.

I get the same error in 1.5 and 1.6-RC4.

**Comment by SentryMan** [comment]
nice catch, it seems this part was overlooked. While I'm fixing it, you can add a string field to one of the types and it'll work.

**Comment by rob-bygrave** [comment]
Fixed in version 1.6-RC5 which I am releasing now.

I'm keen to add support for #109 and then release 1.6
